lib/client.js

```diff
@@ -555,6 +555,13 @@ class Parser {
 
     this.llhttp = exports
     this.ptr = this.llhttp.llhttp_alloc(constants.TYPE.RESPONSE)
+
+    // this.llhttp.llhttp_set_lenient_chunked_length(this.ptr, 1)
+    // this.llhttp.llhttp_set_lenient_headers(this.ptr, 1)
+    // this.llhttp.llhttp_set_lenient_keep_alive(this.ptr, 1)
+    // this.llhttp.llhttp_set_lenient_transfer_encoding(this.ptr, 1)
+    this.llhttp.llhttp_set_lenient_spaces_after_chunk_size(this.ptr, 1)
```

Comment:
we could activate all lenien flags. but actually we needed only `llhttp_set_lenient_spaces_after_chunk_size` How about hiding it behind a n optional field?

Reply:
What would be the use cases of it?

Reply:
lenient flag?

Reply:
I would set all of those to `true`. Wdyt @ShogunPanda?

Reply:
I agree on both; let's create a `insecureHTTPParser` flag like we do for node. If enabled, it will set all flags to `true`.

See:

https://github.com/nodejs/node/blob/8a41d9b636be86350cd32847c3f89d327c4f6ff7/lib/_http_client.js#L800
https://github.com/nodejs/node/blob/main/src/node_http_parser.cc#L94C1-L98C63

Reply:
@Uzlopak @mcollina  Do you want to do this or do we land this PR (with no lenient flags set) and I'll do the work in another PR?

Reply:
I changed this PR to only upgrade llhttp. The changes in client.js are now remove.

I deactivated two tests. Are these regressions or is llhttp handling these two cases more gracefully?

If we upgrade llhttp then I think your PR should be following immediatly? We could also park this PR in a different branch in this repo. Then we would not have heavy pressure.

Reply:
The two PR are independent: when talking to HTTP compliant server undici will still work. 
I'll try to get this PR today.